lib/components/package.json

```diff
@@ -52,6 +51,7 @@
     "@popperjs/core": "^2.6.0",
     "@types/color-convert": "^2.0.0",
     "@types/overlayscrollbars": "^1.12.0",
+    "@types/react-syntax-highlighter": "11.0.5",
```

Comment:
The real problem here is that `@types/react-syntax-highlighter` has a dependency on `@types/react` and that should be changed to an **optional** peer dependency.

Reply:
All `@storybook/*` packages is used by end users after TypeScript building process that generating `*.js` and `*.d.ts` files.

We don't need to provide development dependencies, like `@types/react-syntax-highlighter` for the end users not as dependency nor as optional peer dependency.

Can you explain, please, why the end user of the `@storybook/addon-a11y` need an `@types/react-syntax-highlighter` as optional peer dependency and not  as dev dependency in `node_modules`?

Reply:
> Can you explain, please, why the end user of the @storybook/addon-a11y package needed the @types/react-syntax-highlighter as optional peer dependency?

I didn't say that. I said that the problem here is that `@types/react-syntax-highlighter` is being published with `@types/react` as its dependency, whereas it should be an optional peer dep. That way a dependency on `@types/react-syntax-highlighter` in `@storybook/components` wouldn't be such a problem.

Reply:
I've rechecked this with the `@storybook/components` package itself and I think that this particular change is OK - I don't see a direct reference to `react-syntax-highlighter` in the generated `.d.ts`. In fact, its types are manually copied over to this file: https://github.com/storybookjs/storybook/blob/8fdb943107e54fd2215aa2cc93179b4a9ebe49ea/lib/components/src/syntaxhighlighter/syntaxhighlighter-types.ts

My point is though that each such change has to be evaluated separately because not all of them might be safe.

Reply:
Ok.